akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -122,23 +125,37 @@ trait FileUploadDirectives {
         import ctx.materializer
         import ctx.executionContext
 
+        val done = Promise[RouteResult]()
+
         // Streamed multipart data must be processed in a certain way, that is, before you can expect the next part you
         // must have fully read the entity of the current part.
         // That means, we cannot just do `formData.parts.runWith(Sink.seq)` and then look for the part we are interested in
         // but instead, we must actively process all the parts, regardless of whether we are interested in the data or not.
```

Comment:
Maybe you can add a new comment here that the promise is needed to continue response processing even when parts of the request are still coming in.